pkg/virt-handler/vm.go

```diff
@@ -1763,9 +1779,14 @@ func (d *VirtualMachineController) processVmUpdate(origVMI *v1.VirtualMachineIns
 			}
 
 			// configure network inside virt-launcher compute container
-			err = d.setPodNetworkPhase1(vmi)
+			criticalNetworkError, err := d.setPodNetworkPhase1(vmi)
```

Comment:
I think that checking if pod network is configured on the VMI should be the first step of the migration process, for the target and for the source VMI's, so in case there are issues with the network we can fail faster (lines 1782-1789 should be before 1776)

Reply:
Can you please explain why do you think the order should be changed? We don't know whether we will have an error with mounting the disk or setting the network.

Reply:
First It make more sense to first check if there is network connectivity between the target and the source, which is the infrastructure to the whole migration process.
It is more likely to fail on network connectivity thus you can stop the migration processes earlier if there is a problem.
It is like you would to automate file copy  two computers on the same network, 
first thing you should do is checking that there is connectivity between those computers before anything else, this is part of the sanity check (after checking that the file you want to copy actualy there)

Reply:
I still don't understand why checking the network should be done before checking the disks.
Regarding your example copying the file. Is checking the file exists should be done before or after the network? Is checking the target computer has enough space for the file should be done before or after? I don't know what is the correct order and whether there is one, but I don't think it is a tautology that networking should be the first thing to be checked.
2. We are not doing a sanity check here. We are actually configuring the network.
3. I don't think it really matters which one of those two (network or disks) would be done before. Since the disks are already before the network, in the scope of this PR I prefer to leave at as it.